spec/Section 3 -- Type System.md

```diff
@@ -917,6 +922,30 @@ May yield the result:
 The type of an object field argument must be an input type (any type except an
 Object, Interface, or Union type).
 
+**Oneof Fields**
+
+Oneof Fields are a special variant of Object Type fields where the type system
+asserts that exactly one of the field's arguments must be set and non-null, all
+others being omitted. This is useful for representing situations where an input
+may be one of many different options.
```

Comment:
this isn't exactly wrong, but "may" is often an RFC term of art

"must be exactly one of many" is probably clearer and more RFC-y?